src/Libraries/Microsoft.Extensions.AI.Abstractions/Embeddings/EmbeddingGeneratorMetadata.cs

```diff
@@ -35,17 +36,19 @@ public EmbeddingGeneratorMetadata(string? providerName = null, Uri? providerUri
     /// <summary>Gets the URL for accessing the embedding generation provider.</summary>
     public Uri? ProviderUri { get; }
 
-    /// <summary>Gets the ID of the model used by this embedding generation provider.</summary>
+    /// <summary>Gets the ID of the default model used by this embedding generator.</summary>
     /// <remarks>
-    /// This value can be null if either the name is unknown or there are multiple possible models associated with this instance.
+    /// This value can be null if no default model is set on the corresponding embedding generator.
     /// An individual request may override this value via <see cref="EmbeddingGenerationOptions.ModelId"/>.
     /// </remarks>
-    public string? ModelId { get; }
-
-    /// <summary>Gets the number of dimensions in the embeddings produced by this instance.</summary>
-    /// <remarks>
-    /// This value can be null if either the number of dimensions is unknown or there are multiple possible lengths associated with this instance.
-    /// An individual request may override this value via <see cref="EmbeddingGenerationOptions.Dimensions"/>.
-    /// </remarks>
-    public int? Dimensions { get; }
+    public string? DefaultModelId { get; }
+
+    /// <summary>
+    /// Gets metadata for the the default model or a specified model.
+    /// </summary>
+    /// <param name="modelId">The model identifier. If not set, uses <see cref="DefaultModelId"/>.</param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
+    /// <returns>A task that completes with the model metadata.</returns>
+    public virtual Task<EmbeddingModelMetadata> GetModelMetadataAsync(string? modelId = null, CancellationToken cancellationToken = default)
+        => Task.FromResult(_emptyModelMetadata);
```

Comment:
Can we cache the `Task<Metadata>` rather than just the `Metadata`?

Reply:
Good point. Done.